packages/store/src/state-factory.ts

```diff
@@ -103,15 +108,25 @@ export class StateFactory {
       if (actionMetas) {
         for (const actionMeta of actionMetas) {
           const stateContext = this.createStateContext(getState, setState, dispatch, metadata);
-          const result = metadata.instance[actionMeta.fn](stateContext, action);
-          if (result) {
+          let result = metadata.instance[actionMeta.fn](stateContext, action);
+          if (result === undefined) {
+            result = of({}).pipe(shareReplay());
+          } else if (result instanceof Promise) {
+            result = fromPromise(result);
+          }
+
+          if (result instanceof Observable) {
+            result = result.pipe(
+              actionMeta.options.cancellable ? takeUntil(actions$.pipe(ofAction(action.type))) : map(r => r)
+            ); // act like a noop
+
             results.push(result);
```

Comment:
it looks like results are only being pushed into the array if it is an observable? I am assuming it should always push into the results array?